gamehub-backend/src/test/java/com/example/gamehubbackend/services/GameServiceUnitTests.java

```diff
@@ -0,0 +1,129 @@
+
+        verify(gameRepo).findAll();
+        assertEquals(expectedGames, actualGames);
+    }
+
+    @Test
+    void getAllGames_WhenEmpty_ReturnsEmptyList() {
+        List<Game> allGames = List.of();
+        List<Game> actualGames = gameService.getAllGames();
+
+        assertEquals(allGames, actualGames);
+    }
+
+    @Test
+    void getGameById_Test() {
+        Game game = new Game("1","Super Mario World", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123", "linkToImg");
+        when(gameRepo.findById("1")).thenReturn(Optional.of(game));
+
+        Game actualGame = gameService.getGameById("1");
+
+        Game expectedGame = new Game("1","Super Mario World", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123", "linkToImg");
+        verify(gameRepo).findById("1");
+        assertEquals(expectedGame, actualGame);
+    }
+
+    @Test
+    void getGame_Test_whenGameDoesNotExist_ThenThrowException() {
+        when(gameRepo.findById("1")).thenReturn(Optional.empty());
+
+        assertThrows(GameNotFoundException.class, () -> gameService.getGameById("1"));
+        verify(gameRepo).findById("1");
+    }
+
+    @Test
+    void addGame_Test_whenGameAsInput_thenReturnNewGame() {
+        GameDTO gameDTO = new GameDTO("Super Mario World", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123", "linkToImg");
+        Game gameToSave = new Game("1","Super Mario World", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123", "linkToImg");
+
+        when(gameRepo.save(gameToSave)).thenReturn(gameToSave);
+        when(idService.randomId()).thenReturn(gameToSave.id());
+
+        Game actualGame = gameService.addGame(gameDTO);
+        Game expectedGame = new Game("1","Super Mario World", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123", "linkToImg");
+
+        verify(gameRepo).save(gameToSave);
+        verify(idService).randomId();
+        assertEquals(expectedGame, actualGame);
+    }
+
+    @Test
+    void updateGame_Success_Test() {
+        String id = "1";
+
+        Game existingGame = new Game("1","Super Mario World", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123", "linkToImg");
+        GameDTO updateGameDTO = new GameDTO("Super Mario World 2", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123456", "linkToImg");
+        Game updateGame = new Game("1","Super Mario World 2", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123456", "linkToImg");
+
+        when(gameRepo.findById(id)).thenReturn(Optional.of(existingGame));
+        when(gameRepo.save(updateGame)).thenReturn(updateGame);
+
+        Game actual = gameService.updateGame(updateGameDTO,id);
+
+        assertNotNull(actual);
+        assertEquals(updateGame,actual);
```

Comment:
i think the expected value here should be a new object again, rather than the one you are saying should be returned with line 94?
```suggestion
         Game expected = new Game("1","Super Mario World 2", "Jump and Run", localDate, List.of("NES","SNES"), "Nintendo", "Nintendo", "test123456", "linkToImg");
         assertEquals(expected, actual);
```

Reply:
thanks, fixed  :)